Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.cpp

```diff
@@ -103,7 +103,8 @@ void WebFrameLoaderClient::dispatchDecidePolicyForNavigationAction(const Navigat
         { },
         WTFMove(originatingFrameID),
         WTFMove(parentFrameID),
-        getCurrentProcessID()
+        getCurrentProcessID(),
+        navigationAction.initiatedByMainFrame() == InitiatedByMainFrame::Yes
```

Comment:
This should do the same thing as WebFrame::info, but with requestingFrame if it's non-null.  Make WebFrame::isFocused() to share code in the 2 locations.